codetransformer/assembler.py

```diff
@@ -0,0 +1,99 @@
+import sys
+import types
+from toolz import mapcat
+
+from . import instructions as instrs
+from .code import Code
+
+
+Label = instrs.Label
+
+
+def assemble_function(signature, objs, code_kwargs=None, function_kwargs=None):
```

Comment:
We need to decide how we want to manage signatures here.

Ideally it should be easy to either type this by hand or use the signature of an existing function.  For the tests, I've been using `inspect.signature(lambda <sig>: None)` as an easy way to get the signature I want.

Reply:
one idea is to allow a function which would likely look like: `lambda a, b, c: ...`